osu.Server.Queues.ScoreStatisticsProcessor/Stores/BeatmapStore.cs

```diff
@@ -107,12 +131,40 @@ public static async Task<BeatmapStore> CreateAsync(MySqlConnection connection, M
             return difficultyAttributes;
         }
 
+        /// <remarks>
+        /// This method attempts to create a simple solution to deciding if a <see cref="Mod"/> can be considered a ranked "legacy" mod.
+        /// Used by <see cref="GetDifficultyAttributesAsync"/> to decide if the current mod combination's difficulty attributes
+        /// can be fetched from the database.
+        /// </remarks>
+        private static bool isRankedLegacyMod(Mod mod) =>
```

Comment:
I guess this is the real question of this PR is whether this is what we want. I'm pretty neutral on it, just have to add it to the list of things in our ever-increasing mental map of what needs to be updated when we make changes or add new mods.

I do agree with this listing legacy mods though. It's less likely that we'll add new mods to this list (will probably only be a consideration when we notice the pp calculations running slowly and need to force more performance out of things).

Reply:
I'm not even sure how to cross-check that this list is correct / complete right now.

Also I'd probably slightly prefer if this was listing leaf classes in the inheritance classes (don't like the `// this also catches ManiaModFadeIn` comment, and worried about future accidental inheritance), but probably not as important as _not being to tell if this is correct_.

Reply:
There are two places to check (that I won't link to, for hopefully obvious reasons). So at least I know that now.

Reply:
Now doing more precise checks to aid against accidental inheritance.

Reply:
You fixed the one case that was a problem but I still was not confident there won't be future problems with this so I pushed a test case (https://github.com/ppy/osu-queue-score-statistics/pull/274/commits/327669a1ec444100ade5444020d0fa9e44aac7e8) which should hopefully fail if someone inherits wrong in the future.